### ISSUE TITLE
[REQ] Component Request: Multiple DS18X20 on same 1-wire bus

### ISSUE THREAD

**Issue by laslowh** [adafruit/Wippersnapper_Components#122]
**What's the full name of the component you'd like to see in Adafruit.io WipperSnapper?**
DS18X20 (basic support is included already)

**What type of component is this (i.e.: Pin, I2C)?**
1-wire

**Additional context**
Since these devices are 1-Wire devices, they can be configured as a bus on the same IO pin.  However, I think the component is currently only set up to read the temperature from the first of these devices.  It would be nice for the component to report all of them.

**Comment by brentru** [comment]
The way the API is configured, there is only one OneWire device allowed **per pin**.  You can use multiple DS18x20 devices by using multiple pins.

**Comment by laslowh** [comment]
Understood.  And that is a reasonable workaround.  I think it still would be good not to require eating up IO pins unnecessarily.

**Comment by brentru** [comment]
Absolutely! It was too difficult at the time, possibly we'll revisit this. I am leaving this issue open and have marked it as an enhancement.

**Comment by tyeth** [comment]
Mentioning this as a reminder for our API V2 rewrite, to allow multiple DS18b20 / OneWire devices on the same addresses (currently a pin only component in API v1).
 This also extends to include the idea of I2C OneWire mux chips, like the DS2484 being added in #228, which would also ideally support multiple OneWire devices on their buses.

**Comment by tyeth** [comment]
Just to add it might be worth including a serial number subcomponent option for DS18b20-like components, so sensor swaps can be determined.

For full multi-DS18b20 support we'd need to refactor to consider the onewire as a bus interface supporting multiple components (that cannot be pre-addressed by broker and must be experimentally determined).
Presumably one would end up with feeds like ws-ds18b20-p13-sn000000001 for a pin13 onewire bus with a single sensor returned (serial 1), and the broker would automatically accept/create as needed. This runs against the free feed limit issue though, so far from ideal. 
We also could not easily facilitate both serial and temp in one datapoint (we could as JSON but then unusable by users in IO web + graphs etc).
